src/platform/Darwin/DnssdImpl.cpp

```diff
@@ -450,6 +471,34 @@ CHIP_ERROR ChipDnssdResolve(DnssdService * service, chip::Inet::InterfaceId inte
     return Resolve(context, callback, interfaceId, service->mAddressType, regtype.c_str(), service->mName);
 }
 
+void ChipDnssdResolveNoLongerNeeded(const char * instanceName)
+{
+    auto existingCtx = MdnsContexts::GetInstance().GetExistingResolveForInstanceName(instanceName);
+    if (!existingCtx)
+    {
+        return;
+    }
+
+    if (*existingCtx->consumerCounter == 0)
```

Comment:
This should not happens right ?

Reply:
Ah, this is about the other check, the one before we decrement the counter.

This _could_ happen with some weird and unexpected access patterns: if someone starts a new resolve for an instance name while being notified about a completed resolve for that instance name.  The guard is there to ensure we don't leak, at least, in that situation.